index_configuration_tool/tests/test_main.py

```diff
@@ -0,0 +1,172 @@
+    return config_section
+
+
+class TestMain(unittest.TestCase):
+    # Run before each test
+    def setUp(self) -> None:
+        with open(LOGSTASH_EXPECTED_OUTPUT_FILE, "rb") as f:
+            self.loaded_logstash_config = pickle.load(f)
+
+    def test_get_auth(self):
+        # Empty input should produce empty auth tuple
+        self.assertFalse(main.get_auth(dict()))
+        base_data = {"user": "user", "password": "password"}
+        # Password without user should return empty auth tuple
+        test_data = dict(base_data)
+        del test_data["user"]
+        self.assertFalse(main.get_auth(test_data))
+        # User without password should return empty auth tuple
+        test_data = dict(base_data)
+        del test_data["password"]
+        self.assertFalse(main.get_auth(test_data))
+        # Test valid input
+        result = main.get_auth(base_data)
+        assert type(result) is tuple
+        self.assertEqual("user", result[0])
+        self.assertEqual("password", result[1])
+
+    def test_get_endpoint_info(self):
+        # Simple base case
+        test_config = create_plugin_config(["test"])
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("http://test/", result[0])
+        self.assertFalse(result[1])
+        # SSL enabled
+        test_config = create_plugin_config(["test"], True)
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("https://test/", result[0])
+        self.assertFalse(result[1])
+        # SSL disabled, invalid auth config
+        test_config = create_plugin_config(["test"], False, "test")
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("http://test/", result[0])
+        self.assertFalse(result[1])
+        # SSL disabled, valid auth config
+        test_config = create_plugin_config(["test"], user="user", password="password")
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("http://test/", result[0])
+        self.assertTrue(result[1])
+        # Array of hosts uses the first entry
+        test_config = create_plugin_config(["test1", "test2"], True, "user", "password")
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("https://test1/", result[0])
+        self.assertTrue(result[1])
+
+    def test_print_report(self):
+        # Base case should return an empty list
+        self.assertFalse(main.print_report(dict(), dict()))
+        base_data = {
+            "index1": {
+                "settings": {
+                    "bool_key": True,
+                    "num_key": 1
+                },
+                "mappings": {
+                    "str_key": "value"
+                }
+            },
+            "index2": {}
+        }
+        result = main.print_report(base_data, dict())
+        self.assertEqual(2, len(result))
+        self.assertTrue("index1" in result)
+        self.assertTrue("index2" in result)
```

Comment:
more a clarification question: earlier you were using vanilla `assert`s -- is there any difference in the use cases or advantage to using `assertTrue`?

Reply:
I went down a rabbit hole on this, but here's the short version:

1. `self.assert*` is better than Python's `assert`
2. `self.assertEqual` is better than `self.assertTrue / self.assertFalse`

The reason for (1) is because the `self.assert*` methods return an output object that you can inspect for more granular assertions. Meanwhile, the `assert` statement would simply output an `AssertionError` with no other context.

The reason for (2) is because the failure message for `assertTrue / assertFalse` is simply `False is not True` which is not particularly useful, while `assertEqual` will actually print the mismatched values. See [this StackOverflow answer](https://stackoverflow.com/a/45947566) for more.

Keeping this in mind, i've updated the unit tests to use `assertEqual` wherever possible